app/parse.py

```diff
@@ -1,12 +1,160 @@
+import logging
+import time
 from dataclasses import dataclass
+from functools import wraps
+from typing import Any
+
+from aiohttp import ClientSession, ClientTimeout
+from bs4 import BeautifulSoup
+import asyncio
+
+BASE_URL = "https://mate.academy/"
 
 
 @dataclass
 class Course:
     name: str
     short_description: str
     duration: str
+    module_count: int
+    topics: list[tuple[str, str]]
+
+
+def configure_logging() -> None:
+    logging.basicConfig(
+        level=logging.DEBUG,
+        format="%(asctime)s - %(levelname)s - %(message)s",
+        handlers=[logging.FileHandler("app.log"), logging.StreamHandler()],
+    )
+
+
+def log_time(func: callable) -> Any:
```

Comment:
The type hint for `func` in `log_time` should be `Callable[..., Awaitable[Any]]` instead of `callable` to indicate that it wraps an async function.